### PR TITLE
Feature/registration service

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
Add registration service implementation.
Description is here https://soramitsu.atlassian.net/wiki/spaces/D3/pages/491421721/Client+registration

### CODE REVIEW

File: src/main/kotlin/registration/EthFreeWalletsProvider.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package registration
+
+/**
+ * Provides with free ethereum relay wallet
+ */
+class EthFreeWalletsProvider {
+
+    // TODO add effective implementation
```
Comment (luckychess) [inline src/main/kotlin/registration/EthFreeWalletsProvider.kt]: Fix TODO plz.

File: src/main/kotlin/registration/RegistrationStrategyImpl.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    /**
+     * Register new notary client
+     * - CreateAccount with client name
+     * - SetAccountDetail on client account with assigned relay wallet from notary pool of free relay addresses
+     * - SetAccountDetail on notary node account to mark relay address in pool as assigned to the particular user
+     * @param name - client name
+     * @param pubkey - client public key
+     */
+    override fun register(name: String, pubkey: String): Result<Unit, Exception> {
```
Comment (luckychess) [inline src/main/kotlin/registration/RegistrationStrategyImpl.kt]: Please check what happens if stateful validation failed for one of iroha transactions.

File: src/main/kotlin/registration/RegistrationServiceEndpoint.kt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ */
+class RegistrationServiceEndpoint(
+    port: Int,
+    private val registrationStrategy: RegistrationStrategy
+) {
+
+    init {
+        val server = embeddedServer(Netty, port = port) {
+            routing {
+                post("/register") {
```
Comment (Alexey-N-Chernyshov) [inline src/main/kotlin/registration/RegistrationServiceEndpoint.kt]: not RESTful enough, rename to `/user`

File: src/main/kotlin/registration/RegistrationStrategyImpl.kt (hunk 2 above)
Comment (Alexey-N-Chernyshov) [reply src/main/kotlin/registration/RegistrationStrategyImpl.kt]: done

**Pull request merged**